---
# Documentation: https://wowchemy.com/docs/managing-content/

title: "New approaches for using gene order data in phylogeny reconstruction"
authors: ["Robert K. Jansen", "David Bader", "Bernard M.E. Moret", "Linda A. Raubeson", "Li-San Wang", "Tandy Warnow", "Stacia Wyman"]
date: 2001-08-12T21:24:52-04:00
doi: ""

# Schedule page publish date (NOT publication's date).
publishDate: 2021-08-24T21:24:52-04:00

# Publication type.
# Legend: 0 = Uncategorized; 1 = Conference paper; 2 = Journal article;
# 3 = Preprint / Working Paper; 4 = Report; 5 = Book; 6 = Book section;
# 7 = Thesis; 8 = Patent
publication_types: ["paper-conference"]

# Publication name and optional abbreviated publication name.
publication: "Botany 2001"
publication_short: ""

abstract: "The rapid accumulation of whole genome sequences for a wide diversity
of taxa is generating a huge amount of comparative data for
biologists. The availability of whole genome sequences is providing a
new set of molecular characters for phylogenetic reconstruct ion,
which are especially useful for resolving deep branches of the tree of
life. Changes in gene order are caused primarily by inversions,
transpositions, and transverions.  the major challenges for using
genomic changes is the development of comput ational methods for
handling these types of characters, especially in groups with large
numbers of genes and highly rearranged genomes.  been developing and
testing a variety of methods for reconstructing phylogenies based on
gene order data using both real and synthetic data.  wo of the primary
methods we have developed and tested are Maximum Parsimony on Binary
Encodings (MPBE) and methods for correcting previous published
distance measures (inversion and transposition distances). Our
similations show that all methods perform very well when the rates of
change are low relative to the number of genes and that all methods
perform poorly when rates of change are high relative to the number of
genes.  Furthermore, corrected distance measures greatly improve the
accuracy of phylogenies. We have applied these new methods to a data
set for the highly rearranged chloroplast genomes of the
Campanulaceae. In this group, which generally has low rates of change
relative to the number of genes, all methods recover congruent tree
topologies.  the Campanulaceae are considered accurate because they
are largely congruent with trees generated from three chloroplast gene
sequences (atpB, matk, and rbCL)"

# Summary. An optional shortened abstract.
summary: ""

tags: []
categories: []
featured: false

# Custom links (optional).
#   Uncomment and edit lines below to show custom links.
# links:
# - name: Follow
#   url: https://twitter.com
#   icon_pack: fab
#   icon: twitter

url_pdf:
url_code:
url_dataset:
url_poster:
url_project:
url_slides:
url_source:
url_video:

# Featured image
# To use, add an image named `featured.jpg/png` to your page's folder. 
# Focal points: Smart, Center, TopLeft, Top, TopRight, Left, Right, BottomLeft, Bottom, BottomRight.
image:
  caption: ""
  focal_point: ""
  preview_only: false

# Associated Projects (optional).
#   Associate this publication with one or more of your projects.
#   Simply enter your project's folder or file name without extension.
#   E.g. `internal-project` references `content/project/internal-project/index.md`.
#   Otherwise, set `projects: []`.
projects: []

# Slides (optional).
#   Associate this publication with Markdown slides.
#   Simply enter your slide deck's filename without extension.
#   E.g. `slides: "example"` references `content/slides/example/index.md`.
#   Otherwise, set `slides: ""`.
slides: ""
---

