---
# Documentation: https://sourcethemes.com/academic/docs/managing-content/

title: "UNM Engineering Professors receive $1.1 million in NSF Grants"
subtitle: ""
summary: ""
authors: []
tags: []
categories: []
date: 2001-10-11T06:37:29-04:00
lastmod: 2001-10-11T06:37:29-04:00
featured: false
draft: false

# Featured image
# To use, add an image named `featured.jpg/png` to your page's folder.
# Focal points: Smart, Center, TopLeft, Top, TopRight, Left, Right, BottomLeft, Bottom, BottomRight.
image:
  caption: ""
  focal_point: ""
  preview_only: false

# Projects (optional).
#   Associate this post with one or more of your projects.
#   Simply enter your project's folder or file name without extension.
#   E.g. `projects = ["internal-project"]` references `content/project/deep-learning/index.md`.
#   Otherwise, set `projects = []`.
projects: []
---

[UNM Mirage](200202-UNM-Mirage.pdf)

University of New Mexico School of Engineering Professors Bernard Moret, Computer Science, and **David Bader**, Electrical and Computer Engineering, have received more than $1.1 million in grants this fall from the National Science Foundation (NSF) to pursue research in reconstructing evolutionary trees (known as “phylogenies”). 

This research program is being conducted in collaboration with the University of Texas at Austin. The UT-UNM group received two other awards from the NSF, made directly to the University of Texas, bringing the total group funding for the next five years to more than $7 million. This makes the joint UT-UNM research group the largest and most active group in the world in the area of computational phylogenetics. 

A phylogeny, or evolutionary tree, is a reconstruction of the evolutionary history of a collection of present-day organisms from unknown ancestral organisms through the process of bifurcation, in which an ancestral species gives rise to two new species. Such reconstructions are based on molecular data such as DNA sequences collected from present-day species and on a hypothesized model of evolution.

Moret said the project will create tools for biologists to conduct research on evolution. He said that phylogenetic analyses are used in biological, medical and pharmaceutical research. 

“For instance, public health researchers use phylogenies to track the spread of various strains of the HIV virus (the cause of AIDS). Pharmaceutical companies use them to identify likely drug targets. A notable success was the herbicide Roundup,” he said, adding that agricultural laboratories use them to produce better strains of basic foods such as rice or wheat.

Moret said that the UNM team focuses on the development, implementation and testing of computer algorithms to reconstruct phylogenies, with particular emphasis on efficiency and high-performance computing—because many of the algorithms in use today can require years of computation even on problems of modest size.

Bader said that high-performance computers, which are now used for time-sensitive applications such as weather forecasting, financial investment decision-making, or tracking and guidance systems, will be used for these computational biology problems. 

“The shortened turnaround time will speed up the pace of research, leading to more and less expensive new drugs, for instance, and also improve its quality by enabling researchers to test more alternatives and thus gain a better understanding of the problems,” Bader said.

The $1.1 million is broken up into three separate grants, which deal with different aspects of the project. Two of the grants were awarded under the competitive Information Technology Research (ITR) programs.

A three-year grant for $162,000 titled, “Computing Optimal Phylogenetic Trees under Genome Rearrangement Metrics,” was one in 1,295 proposals received by NSF in its ITR category—only 10 percent of these proposals were funded. Under this and the matching UT grant, Moret, with Robert Jansen, UT biology professor, and Tandy Warnow, UT computer science professor, will design, implement and evaluate new algorithms to reconstruct phylogenies from a new type of molecular data, the complete ordering of genes along the chromosomes.

A five-year grant for $793,000 titled, “Reconstructing Complex Evolutionary Histories,” was one in 661 pre-proposals in its ITR category received by NSF; again only 10 percent of these pre-proposals were eventually selected for funding. Under this and the matching UT grant, Moret and Bader will collaborate with Warnow and several UT-Austin biologists to tackle complex and more realistic models of evolution and develop algorithms that can reconstruct phylogenies under these models for very large problems.

A five-year grant for $193,000 is part of a larger grant titled, “Comparative Chloroplast Genomics: Integrating Computational Methods, Molecular Evolution, and Phylogeny,” awarded to UT- Austin under an NSF program on Biocomplexity in the Environment. The UNM team led by Moret will help the biologists at UT Austin and other institutions in analyzing new genomic data collected from a sampling of land plants and refined into complete gene orderings at the Department of Energy’s Joing Genome Institute in California.

Moret said that funding from NSF is the “gold standard” in the scientific community. All proposals undergo rigorous peer evaluation and fierce competition, as indicated by the funding rate of 10 percent. “Receiving five separate awards in the same year from the NSF for work in the same area is nearly unheard of and a tribute to the success of the UT-UNM collaboration of computer scientists and biologists,” Moret said.

At UNM, these grants will enhance the profile of the research laboratory run by Moret and Bader, which is already funded by four other NSF awards. This includes a three-year ITR award to Moret and Bader granted last year, a five-year CAREER Award to Bader, a three-year biology award in which Bader is a co-PI, and part of a research infrastructure award to the Department of Computer Science.


[http://www.unm.edu/news/Releases/Oct11nsf.htm](https://web.archive.org/web/20011124182239/http://www.unm.edu/news/Releases/Oct11nsf.htm)